libs/onboarding-sdk-crypto/src/main/java/it/pagopa/selfcare/onboarding/crypto/client/NamirialHttpClient.java

```diff
@@ -0,0 +1,72 @@
+package it.pagopa.selfcare.onboarding.crypto.client;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.api.client.http.*;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import it.pagopa.selfcare.onboarding.crypto.entity.SignRequest;
+import org.apache.pdfbox.io.IOUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class NamirialHttpClient {
+
+    private static final String NAMIRIAL_BASE_URL = "https://sws.test.namirialtsp.com/SignEngineWeb";
+
+    private static final String NAMIRIAL_SIGN_PADES_URL = NAMIRIAL_BASE_URL + "/rest/service/signPAdES";
+
+    public byte[] signDocument(SignRequest request) throws IOException {
+        // Initialize HTTP Transport and Request Factory
+        HttpTransport httpTransport = new NetHttpTransport();
+        HttpRequestFactory requestFactory = httpTransport.createRequestFactory();
+
+        // Create the multipart content
+        MultipartContent multipartContent = new MultipartContent();
+        ObjectMapper objectMapper = new ObjectMapper();
+
+
+        String credentials = objectMapper.writeValueAsString(request.getCredentials());
+        String preferences = objectMapper.writeValueAsString(request.getPreferences());
+
+
+        multipartContent.addPart(
+                new MultipartContent.Part(
+                        new HttpHeaders().set("Content-Disposition", "form-data; name=\"file\"; filename=\""+ request.getFile().getAbsolutePath() +"\""),
```

Comment:
Do we need to send the whole file path to Namirial?

```suggestion
                        new HttpHeaders().set("Content-Disposition", "form-data; name=\"file\"; filename=\""+ request.getFile().getName() +"\""),
```

Reply:
fixed it